src/mediapipe_internal/mediapipegraphdefinition.cpp

```diff
@@ -415,4 +434,39 @@ std::pair<std::string, mediapipe_packet_type_enum> MediapipeGraphDefinition::get
     SPDLOG_LOGGER_DEBUG(modelmanager_logger, "setting input stream: {} packet type: {} from: {}", "", "UNKNOWN", streamFullName);
     return {"", mediapipe_packet_type_enum::UNKNOWN};
 }
+
+Status MediapipeGraphDefinition::initializeNodes() {
+#if (PYTHON_DISABLE == 0)
+    SPDLOG_INFO("MediapipeGraphDefinition initializing graph nodes");
+    for (int i = 0; i < config.node().size(); i++) {
+        if (config.node(i).calculator() == PYTHON_NODE_CALCULATOR_NAME) {
+            if (!config.node(i).node_options().size()) {
+                SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Python node missing options in graph: {}. ", this->name);
+                return StatusCode::PYTHON_NODE_MISSING_OPTIONS;
+            }
+            if (config.node(i).name().empty()) {
+                SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Python node name is missing in graph: {}. ", this->name);
+                return StatusCode::PYTHON_NODE_MISSING_NAME;
+            }
+            std::string nodeName = config.node(i).name();
+            if (this->pythonNodeResources.find(nodeName) != this->pythonNodeResources.end()) {
+                SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Python node name: {} already used in graph: {}. ", nodeName, this->name);
+                return StatusCode::PYTHON_NODE_NAME_ALREADY_EXISTS;
+            }
+
+            Status status;
+            std::shared_ptr<PythonNodeResource> state = std::make_shared<PythonNodeResource>(config.node(i).node_options(0), status);
+            if (state == nullptr || status != StatusCode::OK) {
+                SPDLOG_ERROR("Failed to process python node graph {}", this->name);
+                return status;
+            }
+
+            this->pythonNodeResources.insert(std::pair<std::string, std::shared_ptr<PythonNodeResource>>(nodeName, std::move(state)));
```

Comment:
Why not 
`this->pythonNodeResources[nodeName] = state; `
?

Reply:
Insert is good at finding out if the key existed earlier. But the fact is not used here.
```suggestion
            auto inserted, _ = this->pythonNodeResources.insert(std::pair<std::string, std::shared_ptr<PythonNodeResource>>(nodeName, std::move(state)));
            if (!inserted) {
             // error
            }
```
Please handle that

Reply:
Its handled in previous check.